sos/report/plugins/kdump.py

```diff
@@ -124,4 +124,42 @@ def setup(self):
         if self.get_option("collect-kdumps"):
             self.add_copy_spec(["/var/kdump-*"])
 
+
+class AzureKDump(KDump, AzurePlugin):
+
+    files = ('/etc/kdump.conf',)
+    packages = ('kexec-tools',)
+
+    def read_kdump_conffile(self):
+        """ Parse /etc/kdump file """
+        path = "/var/crash"
+
+        kdump = '/etc/kdump.conf'
+        with open(kdump, 'r', encoding='UTF-8') as file:
+            for line in file:
+                if line.startswith("path"):
+                    path = line.split()[1]
+
+        return path
+
+    def setup(self):
+        super().setup()
+
+        self.add_copy_spec([
+            "/etc/kdump.conf",
+            "/usr/lib/udev/rules.d/*kexec.rules"
+        ])
+
+        try:
+            path = self.read_kdump_conffile()
+        except Exception:  # pylint: disable=broad-except
+            # set no filesystem and default path
+            path = "/var/crash"
+
+        if self.get_option("all_logs"):
+            self.add_copy_spec(f"{path}/**")
```

Comment:
This could potentially capture the vmcore themselves, not just the dmesg files. Capturing everything in that directory could create huge sos reports, and I'm not sure what's the benefit to capture old vmcores.

Reply:
We do want to capture the vmcore files. However, it could create really huge sos reports. I took a look at it again and created a `get_vm_core` plugin opt that is set to false by default. If this value is set to true, then we will collect the vmcore only if its size is <= 2GB and has been created in the last 24 hours. What do you think about this approach? :)

Reply:
I like it. In support I remember having the issue of having to ask for both sos report and to upload the vmcores. If we can do this  in one step, we can have all the data we need really easily.

Reply:
Agreed!